### DISCUSSION TITLE
Couldn't find a workspace for URI: error keeps happening

### DISCUSSION THREAD

**Discussion by vitus7** [Shopify/ruby-lsp#2813] [Q&A] [answered]
I've had Ruby 3.3.1 installed using other editors and decided to give Visual Studio Code a try. I installed the Ruby LSP extension but when I try to run/deug code it gives me this error:

Couldn't find a workspace for URI: file:///c%3A/Users/me/nameoffolder or editor: [object Object]

I have been trying to fix this for a while now and can't seem to get it to work.

**Comment by andyw8** [comment]
Does your project have a `Gemfile.lock`?

**Reply by vitus7** [reply]
That was a part of the issue. I had to rebuild everything. Thanks for the help!

**Comment by vinistock** [comment]
The Ruby LSP currently only supports debugging files that are within a VS Code workspace folder. What that message says is that the document you are trying to debug couldn't be associated with any of the workspace folders currently opened in the editor.

Is the file you want to debug an independent script? Where are you opening VS Code in respect to that file?

**Reply by vitus7** [reply]
I see. I'm new to all this so it's a learning process for me. It is an existing file from another directory. I installed VS Code and made new directories to try to start from scratch. I copied the *.rb file over to the new folder. I edited the code and tried to run/debug and got that error.

**Reply by vinistock** [reply]
We don't support usage without a workspace folder opened yet (#1780). If you open VS Code in a directory where your scripts exist and then you try to debug, do you still see the issue?

For example:

```
my_scripts/ << open this directory on VS Code, rather than opening the individual files
  do_something.rb
```

Also, how are you launching the debugger? Launch.json configs? If you still see the issue, can you please share what configs you are using?

**Reply by vitus7** [reply, marked as answer]
It turned out to be more than one thing. I wasn't opening the folder like you said -- I was opening the file directly. It also turned out that my gemfile had issues and needed to be rebuilt. Now it shows Ruby LSP is running in the lower right corner. Now I just need to figure out how to get it to run the script and see the output. Thanks for the help!
